.travis.yml

```
@@ -7,32 +9,29 @@ stages:
 
 if: branch = master OR type = pull_request
 
-stage: test
-
-name: Project Tests
-
-language: python
-
-python:
-  - "3.8"
-  - "3.9"
-
-install: skip
-
-script:
-  - |
-    set -e
-    make docker/run target=python/test
-    make localstack/up
-    make localstack/pytest
-    make localstack/clean
+before_install:
+  - tmpdaemon=$(mktemp)
+  - sudo jq '."registry-mirrors" += ["https://mirror.gcr.io"]' /etc/docker/daemon.json > "$tmpdaemon"
+  - sudo mv "$tmpdaemon" /etc/docker/daemon.json
+  - sudo systemctl daemon-reload
+  - sudo systemctl restart docker
+  - docker system info
 
 jobs:
   include:
     - stage: lint
       name: Project Syntax Verification
+      script: make docker/run target=lint
+    - stage: test
+      name: Run unit and integration tests against a mock AWS stack
+      install:
+        - make docker-compose/install
       script:
-        - make docker/run target=lint
+        - make docker/run target=pytest/lambda/tests
```

The trouble with using a list of commands for the `script` argument is that travis-ci will always run _all_ of them, even if one fails. The job is still marked as a failure, but the output can be confusing because the last command may not be the one that failed.

The `install` target works more intuitively, where the job fails as soon as a command fails, so moving more of the commands there can help.

A couple other ideas... We could use a multi-line block with `set -e`:

```suggestion
      script: |
        set -e
        make docker/run target=pytest/lambda/tests
        make mockstack/up
        make mockstack/pytest/lambda
```

Or we could add a local `make` target in this project that runs all the commands, since the `make` behavior is to fail as soon as any command fails. Then we'd specify just that target here.

---

A third approach would be to separate the unit tests from the integration tests, creating a stage where just the unit tests are run, then a stage where just the integration tests are run.  I don't need mockstack for the unit tests.

That would be my first choice, my second choice would be the "set -e".   I'm not as crazy about grouping them together in a new target for the Makefile.

You decide.

I thought you would complain about the plethora of requirement files.  I tried sticking the files all in a requirements directory, but ran into other issues.  I don't especially like what I have now, but I dislike the alternatives more so.

---

>I thought you would complain about the plethora of requirement files. I tried sticking the files all in a requirements directory, but ran into other issues. I don't especially like what I have now, but I dislike the alternatives more so.

I've done the requirements directory in the past, and that would be my first choice, but it's not critical.

>A third approach would be to separate the unit tests from the integration tests, creating a stage where just the unit tests are run, then a stage where just the integration tests are run. I don't need mockstack for the unit tests. That would be my first choice...

Yep, that would also work. Go for it!